src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonPropertyInfo.cs

```diff
@@ -654,11 +654,25 @@ public string Name
                 }
 
                 _name = value;
+
+                // This setter should only be called by end users.
+                // Disable fast-path if a user modifies a property name.
+                if (ParentTypeInfo != null)
+                {
+                    ParentTypeInfo.CanUseSerializeHandler = false;
+                }
             }
         }
 
         private string? _name;
 
+        internal void SetNameInternal(string name)
```

Comment:
Is this simply to avoid running a null check on `name`? Why is it needed, metadata configuration is hardly on the hot path for the serializer.

Reply:
No it's to make sure that if user's change to a different property name at runtime, we don't go down the fast-path which has the stuff pre-encoded at compile-time.